src/main/java/org/opensearch/knn/index/KNNMethod.java

```diff
@@ -77,6 +78,35 @@ public ValidationException validate(KNNMethodContext knnMethodContext) {
         return validationException;
     }
 
+    public ValidationException validateWithData(KNNMethodContext knnMethodContext, TrainingDataSpec trainingDataSpec) {
+        List<String> errorMessages = new ArrayList<>();
+        if (!containsSpace(knnMethodContext.getSpaceType())) {
```

Comment:
nit: i though you are actually checking for space in string, isSpaceTypeSupported() might be good for readability ( FAR)